tuf/client/updater.py

```diff
@@ -643,6 +643,13 @@ def refresh(self, unsafely_update_root_if_necessary=True):
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
+    # If an exception is raised during the metadata update attempts, we will
+    # attempt to update root metadata once by recursing with a special argument
+    # to avoid further recursion. We use this bool and pull the recursion out
+    # of the except block so as to avoid unprintable nested NoWorkingMirrorError
```

Comment:
Instead of using a second boolean (`unsafely_update_root_if_necessary` being the first one) to take care of retries, would it work to stick the following in the two places that `retry_once` is used?

```Python
self._update_metadata('root', DEFAULT_ROOT_UPPERLENGTH)
self.refresh(unsafely_update_root_if_necessary=False)
```

I find it confusing to use two boolean variables for the same thing, unless its an issue with printing nested exceptions.  If it's the latter, I'd like to fix the source of the problem.